src/platforms/linux-raspbian.js

```diff
@@ -137,7 +137,18 @@ function setHostname(hostname) {
     return false;
   }
 
-  return true;
+  proc = child_process.spawnSync(
+    'sudo',
+    [
+      'sed',
+      '-i',
+      '-E',
+      '-e',
+      `s/(127\\.0\\.0\\.1[ \\t]+)${original}/\\1${hostname}/g`,
+      '/etc/hosts',
+    ]
+  );
+  return proc.status === 0;
```

Comment:
/etc/hostname contains the hostname on the RPi. /etc/hosts should also be updated.
Rather than updating /etc/hostname, it's probably better to call `sudo hostname new-name` (which will update /etc/hostname, but not /etc/hosts).

Reply:
This was just an addition. `/etc/hostname` is already updated inside this function, and `hostname` is also called. However, calling `hostname` doesn't actually persist the new hostname to `/etc/hostname`.

Reply:
And In the /etc/hosts I've seen on the RPi, the address was 127.0.1.1 rather than 127.0.0.1

Here's the raspi-config routine that changes the hostname:
https://github.com/RPi-Distro/raspi-config/blob/1960514ea73d3669c370f9cb4d63c1f532c60784/raspi-config#L480

Reply:
Ah, that's what I based my code off, but I didn't see they used 127.0.1.1.

Reply:
You're right hostname doesn't persist anything. But it does change something internally since if you do `sudo hostname foo` sudo then complains until /etc/hosts gets updated. I guess that's why raspi-config wants a reboot.